src/node/internal/internal_http_server.ts

```diff
@@ -259,7 +271,17 @@ export class Server
     incoming._addHeaderLines(headers, headers.length);
 
     incoming.method = request.method;
-    incoming._stream = request.body;
+
+    // For POST/PUT requests with body, pre-read the entire body to match Node.js behavior.
+    // The bodyReadPromise ensures that request data is buffered before middleware can attach
+    // listeners, preventing Express.js body parsing issues where 'data' events are missed.
```

Comment:
Just need to understand the implication of this a bit more. Is this saying that the entire body payload needs to be read into memory? That seems... unusual... if so.